src/main/java/com/cburch/logisim/gui/menu/MenuEdit.java

```diff
@@ -160,6 +205,17 @@ public void actionPerformed(ActionEvent e) {
         proj.undoAction();
       } else if (src == redo && proj != null) {
         proj.redoAction();
+      } else if (src == clearHistory && proj != null) {
+        final var result = JOptionPane.showConfirmDialog(
+          proj.getFrame(),
+          S.get("clearHistoryWarningMessage"),
+          S.get("clearHistoryWarningTitle"),
+          JOptionPane.OK_CANCEL_OPTION,
+          JOptionPane.WARNING_MESSAGE
+        );
+        if (result == JOptionPane.OK_OPTION) {
+            proj.discardAllEdits();
```

Comment:
Your code is inconsistently indented. Sometimes you indent 2 spaces, other times 4.

Our style is to indent 2 spaces for code inside { }. We indent at least 4 spaces (and normally 4) for line wraps including parenthesized lists.